pulpcore/cli/core/config.py

```diff
@@ -0,0 +1,117 @@
+from pathlib import Path
+from typing import Optional
+
+import click
+import toml
+
+from pulpcore.cli.common import config_options
+
+LOCATION = str(Path(click.utils.get_app_dir("pulp"), "settings.toml"))
+SETTINGS = ["base_url", "username", "password", "cert", "key", "verify_ssl", "format"]
+
+
+@click.group(name="config", help="Manage pulp-cli config file")
+def config() -> None:
+    pass
+
+
+@config.command(help="Create a pulp-cli config settings file")
+@config_options
+@click.option("--interactive", "-i", is_flag=True)
+@click.option("--editor", "-e", is_flag=True, help="Edit the config file in an editor")
+@click.option("--overwrite", "-o", is_flag=True, help="Overwite any existing config file")
+@click.option("--location", default=LOCATION, type=click.Path(resolve_path=True))
+@click.option(
+    "--format", type=click.Choice(["json", "yaml", "none"], case_sensitive=False), default="json"
+)
+@click.pass_context
+def create(
+    ctx: click.Context,
+    interactive: bool,
+    editor: bool,
+    overwrite: bool,
+    location: str,
+    base_url: str,
+    username: Optional[str],
+    password: Optional[str],
+    cert: Optional[str],
+    key: Optional[str],
+    verify_ssl: bool,
+    format: str,
+) -> None:
+    def _check_location(loc):
+        if not overwrite and Path(loc).exists():
+            raise click.ClickException(
+                f"File '{loc}' already exists. Use --overwite if you want to overwrite it."
+            )
+
+    def prompt_config_option(name):
+        """Find the option from the root command and prompt."""
+        option = next(p for p in ctx.command.params if p.name == name)
+        return click.prompt(option.help, default=(option.default or ""), type=option.type)
+
+    settings = {}
+    if interactive:
+        location = click.prompt("Config file location", default=LOCATION)
+        _check_location(location)
+        for setting in SETTINGS:
+            settings[setting] = prompt_config_option(setting)
+    else:
+        _check_location(location)
+        for setting in SETTINGS:
+            settings[setting] = locals()[setting] or ""
+
+    output = f"[cli]\n{toml.dumps(settings)}"
```

Comment:
This looks hacky to me, not being a toml expert.
Would this also work?
```suggestion
    output = toml.dumps({"cli": settings})
```

Reply:
I must have looked at a bad example as this does work.